### PR TITLE
Tweaked output for kotlin-website samples feature

### PR DESCRIPTION
**Pull request by semoro**
Now all sample blocks in kotlin-website
<div class="sample" markdown="1">...</div>

And small refactoring

### CODE REVIEW

File: core/src/main/kotlin/Samples/SampleProcessingService.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package Samples
```
Comment (yole) [inline core/src/main/kotlin/Samples/SampleProcessingService.kt]: org.jetbrains.dokka.Samples please

File: core/src/main/kotlin/Kotlin/DescriptorDocumentationParser.kt [hunk 2]
```diff
@@ -178,7 +179,18 @@ class DescriptorDocumentationParser
         val lines = text.trimEnd().split("\n".toRegex()).toTypedArray().filterNot(String::isEmpty)
         val indent = lines.map { it.takeWhile(Char::isWhitespace).count() }.min() ?: 0
         val finalText = lines.map { it.drop(indent) }.joinToString("\n")
-        return ContentBlockSampleCode().let { it.append(ContentText(finalText)); it }
+
+        val psiFile = psiElement.containingFile
+        val importsBlock = if (psiFile is KtFile) {
+            ContentBlockCode("kotlin").apply {
+                append(ContentText(psiFile.importList?.text ?: ""))
```
Comment (yole) [inline core/src/main/kotlin/Kotlin/DescriptorDocumentationParser.kt]: Please provide a test for rendering of import lists

File: core/src/main/kotlin/Samples/KotlinWebsiteSampleProcessingService.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+
+    fun buildReplacementData(psiElement: PsiElement): Map<TextRange, String> {
+
+        val result = TreeMap<TextRange, String>({ o1, o2 -> o1.startOffset.compareTo(o2.startOffset) })
+        fun convertAssertPrints(expression: KtCallExpression) {
+            val (argument, commentArgument) = expression.valueArguments
+            val comment = commentArgument.getArgumentExpression() as KtStringTemplateExpression
+            val commentText = comment.entries.joinToString("") { it.text }
+            result[expression.textRange] = "println(${argument.text}) //$commentText"
```
Comment (yole) [inline core/src/main/kotlin/Samples/KotlinWebsiteSampleProcessingService.kt]: Space before comment please

File: core/testdata/format/website/sampleWithAsserts.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+fun a(): String {
+    return "Hello, Work"
+}
+
+fun b(): String {
+    return "Hello, Rest"
+}
+
+fun sample() {
+    assertPrints(a(), "Hello, Work")
```
Comment (yole) [inline core/testdata/format/website/sampleWithAsserts.kt]: Assert methods usually take expected text as first argument.

File: core/src/main/kotlin/Samples/KotlinWebsiteSampleProcessingService.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import java.util.*
+
+open class KotlinWebsiteSampleProcessingService
+@Inject constructor(options: DocumentationOptions,
+                    logger: DokkaLogger,
+                    resolutionFacade: DokkaResolutionFacade)
+    : DefaultSampleProcessingService(options, logger, resolutionFacade) {
+
+
+    fun buildReplacementData(psiElement: PsiElement): Map<TextRange, String> {
```
Comment (yole) [inline core/src/main/kotlin/Samples/KotlinWebsiteSampleProcessingService.kt]: Why do you need such a complicated scheme? Why can't you just go directly over the PSI and produce output text in a single pass, replacing assertions with prints as needed?

File: core/testdata/format/website/sampleWithAsserts.kt (hunk 4 above)
Comment (semoro) [inline core/testdata/format/website/sampleWithAsserts.kt]: In this case, as second, taken from stdlib sample sample

**Pull request merged**